Marlin/Marlin_main.cpp

```diff
@@ -3761,10 +3768,23 @@ inline void gcode_G4() {
     sync_plan_position();
 
     // Move all carriages together linearly until an endstop is hit.
-    current_position[X_AXIS] = current_position[Y_AXIS] = current_position[Z_AXIS] = (Z_MAX_LENGTH + 10);
+    current_position[X_AXIS] = current_position[Y_AXIS] = current_position[Z_AXIS] = (DELTA_HEIGHT + home_offset[Z_AXIS] + 10);
```

Comment:
The reason for this change is that the delta height can be adjusted via the `M665` command, which uses `home_offset[Z_AXIS]` to dynamically adjust the height. The previous use of `Z_MAX_LENGTH` did not account for th adjustment allowed by `M665`. Since this function is only used to home delta machines, I think this is a safe change.